### ISSUE TITLE
pinmame windows/linux SDL2 

### ISSUE THREAD

**Issue by gnulnulf** [vpinball/pinmame#70]
I did a proof of concept for sdl2 in linux and windows.
On linux it works if I let the bitmap-blit.h magic work (with some keyboard issues) .
If I try to directly access bitmap->base I get the correct image with different colors.
Is anyone familiar with the pinmame bitmap object and is it worth the effort?

(if this is the wrong place to ask, what would be the best spot?)

**Comment by toxieainc** [comment]
Hi and sorry, somehow i didn't notice this issue..  :/
Do you have any new insights by now?

**Comment by gnulnulf** [comment]
I got sdl2 somewhat working. ( in my unixsdl2 branch )
There is still a segfault I cannot find but palette shows normal and the debugger is working.

**Comment by toxieainc** [comment]
@gnulnulf Any further news here?

**Comment by gnulnulf** [comment]
Yes, the segfaults were caused by something on my UB20.04 dev instance. 
SDL2 Sound is recognizable but it sounds terrible.

I'm not sure where to start on replacing directX7 with SDL2 on windows.

**Comment by EarBiteR** [comment]
@gnulnulf , You might talk to @francisdb as he did a sdl2 port in Jan that might be relevant - https://github.com/vpinball/pinmame/pull/203

**Comment by francisdb** [comment]
> @gnulnulf , You might talk to @francisdb as he did a sdl2 port in Jan that might be relevant - #203

There's a reason why it's still a draft PR.

This is my first modification of C source code. If I remember correctly there are 2 ways of providing sound to SDL2 and I just picked the simplest model to get going. Also don't plan on working on this in the near future. Feel free to take over.

**Comment by gnulnulf** [comment]
I was distracted by other things ;-)
The linux part works but I think that replacing directX7 with sdl2 would make it more future proof. Personally I use pinmame to test and debug roms.

**Comment by GandalfDG** [comment]
@gnulnulf I've been working on building a simulator for Data East Time Machine, and running the sdl version on Linux. I'd be interested in collaborating on some PinMAME improvements for the linux world since I'm digging into the code now :) I've been trying to find proper channels to contact contributors to the project but it doesn't seem all that active.

Here's what I've been working on https://github.com/GandalfDG/pinmame/tree/time_machine_sim

**Comment by GandalfDG** [comment]
Well, I've just been using the makefile.unixsdl to build, which seems like it's not the currently supported build method. I tried getting things going with cmake, but had issues with nothing actually rendering in the window. For now I'm just trying to get something working to a level for my own debugging use, but I like the project so I'd be interested in helping with overall development as well.

Currently I'm trying to figure out how inports and solenoids are working

**Comment by GandalfDG** [comment]
I've compiled on both Manjaro and Fedora. It might be an issue to do with X11 vs. Wayland. I use KDE plasma as my desktop environment and I could see the game running in the hove preview on the taskbar, but the actual window was blank.

This was with a manual cmake build following the github actions pipeline for reference

**Comment by GandalfDG** [comment]
I'm out of town this weekend, but I'll give this a try and report back when I get a chance.

**Comment by GandalfDG** [comment]
Looks like the francisdb version gives me a output when I build with cmake! @gnulnulf your branch didn't have a linux CMakeLists.txt, so I wasn't able to test that one.

**Comment by gnulnulf** [comment]
Mine still used make. I seem to be the last person to avoid cmake ;-)
Both our trees were a bit aged, I tried to merge it with the current master, francis' sdl2 and my cmake additions for the xpinmamed target. I also added github automation for xpinmamed(linux)

https://github.com/gnulnulf/pinmame/tree/francissdl2

I did not have the time to test it on fedora or alma yet.

**Comment by gnulnulf** [comment]
I cannot seem to find how to make or use the sims. If you like I like to give a try on a cmake migration for your sim.
https://github.com/gnulnulf/pinmame/tree/sims includes your time machine in the linux cmake.

 find build/ |grep tmmc
build/CMakeFiles/xpinmamed.dir/src/wpc/sims/s11/prelim/tmmchn.c.o.d
build/CMakeFiles/xpinmamed.dir/src/wpc/sims/s11/prelim/tmmchn.c.o

**Comment by gnulnulf** [comment]
Right, I mis-merged your sim the first time. now it shows the help and does not complain about missing balls.
Sim is responding here, nice! I tested the build artefact on fedora-42, that works.
